plugins/woocommerce/src/Blocks/Assets/Api.php

```diff
@@ -258,7 +258,7 @@ public function register_script( $handle, $relative_src, $dependencies = [], $ha
 		$script_data = $this->get_script_data( $relative_src, $dependencies );
 
 		if ( in_array( $handle, $script_data['dependencies'], true ) ) {
-			if ( $this->package->feature()->is_development_environment() ) {
+			if ( wp_get_environment_type() === 'development' ) {
```

Comment:
I am not sure how to test this condition, I did add a test case in the PR for the `disable_cache` option.